sql/2021/mobile-web/popular_mobile_input_types.sql

```diff
@@ -0,0 +1,40 @@
+#standardSQL
+# Popular mobile input types
+CREATE TEMPORARY FUNCTION getInputTypes(payload STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+  const almanac = JSON.parse(payload);
+  return almanac.input_elements.nodes.map(function(node) {
+    return node.type.toLowerCase();
+  });
+} catch (e) {
+  return [];
+}
+''';
+
+SELECT
+  total_pages_with_inputs,
+  SUM(COUNT(0)) OVER (PARTITION BY 0) AS total_inputs,
```

Comment:
As you are using `UNNEST` clause this short cut to get the total sites will not be correct. You may have to do a `JOIN` here instead.

Reply:
Same here. I want to be counting the total number of inputs, not the total sites. When counting sites DISTINCT is used, so this shouldnt be a problem

Reply:
Oh in this case it may actually be fine. Did a bit of a blanket sweep so may have over warned. Resolving.

Reply:
Unresolving and advise you to double check as you are doing a `SUM` of `COUNT(DISTINCT url)` which does not necessarily lead to a count of distinct URLs.